packages/generators/lib/create-stackable-files.js

```diff
@@ -0,0 +1,772 @@
+    const config = {
+      $schema: './stackable.schema.json',
+      greeting: {
+        text: this.config.greeting ?? 'Hello world!'
+      }
+    }
+    return Object.assign({}, baseConfig, config)
+  }
+
+  async _afterPrepare () {
+    this.addFile({
+      path: '',
+      file: 'stackable.schema.json',
+      contents: JSON.stringify(schema, null, 2)
+    })
+  }
+}
+
+module.exports = Generator
+module.exports.Generator = Generator
+`
+
+const TS_STACKABLE_GENERATOR_FILE = `\
+import { Generator as ServiceGenerator } from '@platformatic/service'
+import { BaseGenerator } from '@platformatic/generators'
+import { schema } from './schema'
+
+class Generator extends ServiceGenerator {
+  getDefaultConfig (): BaseGenerator.JSONValue {
+    const defaultBaseConfig = super.getDefaultConfig()
+    const defaultConfig = {
+      greeting: 'Hello world!'
+    }
+    return Object.assign({}, defaultBaseConfig, defaultConfig)
+  }
+
+  async _getConfigFileContents (): Promise<BaseGenerator.JSONValue> {
+    const baseConfig = await super._getConfigFileContents()
+    const config = {
+      $schema: './stackable.schema.json',
+      greeting: {
+        text: this.config.greeting ?? 'Hello world!'
+      }
+    }
+    return Object.assign({}, baseConfig, config)
+  }
+
+  async _afterPrepare () {
+    this.addFile({
+      path: '',
+      file: 'stackable.schema.json',
+      contents: JSON.stringify(schema, null, 2)
+    })
+  }
+}
+
+export default Generator
+export { Generator }
+
+`
+
+const JS_STACKABLE_SCHEMA_FILE = `\
+'use strict'
+
+const { schema } = require('@platformatic/service')
+
+const stackableSchema = {
+  ...schema.schema,
+  $id: 'stackable',
+  title: 'Stackable Config',
+  properties: {
+    ...schema.schema.properties,
+    greeting: {
+      type: 'object',
+      properties: {
+        text: {
+          type: 'string'
+        }
+      },
+      required: ['text'],
+      additionalProperties: false
+    }
+  },
+}
+
+module.exports.schema = stackableSchema
+
+if (require.main === module) {
+  console.log(JSON.stringify(stackableSchema, null, 2))
+}
+`
+
+const TS_STACKABLE_SCHEMA_FILE = `\
+import { schema } from '@platformatic/service'
+
+const stackableSchema = {
+  ...schema.schema,
+  $id: 'stackable',
+  title: 'Stackable Config',
+  properties: {
+    ...schema.schema.properties,
+    greeting: {
+      type: 'object',
+      properties: {
+        text: {
+          type: 'string'
+        }
+      },
+      required: ['text'],
+      additionalProperties: false
+    }
+  },
+}
+
+export { stackableSchema as schema }
+
+if (require.main === module) {
+  console.log(JSON.stringify(stackableSchema, null, 2))
+}
+`
+
+const STACKABLE_CONFIG_TYPES_FILE = `\
+/* eslint-disable */
+/**
+ * This file was automatically generated by json-schema-to-typescript.
+ * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
+ * and run json-schema-to-typescript to regenerate this file.
+ */
```

Comment:
This file should be autogenerated by `json-schema-to-typescript`, not by us.

Reply:
Here is a problem. When we generate a ts project:
- to generate a `config.d.ts` we need to compile the project 
- to compile the project we need to have a `config.d.ts`

Reply:
This file can be a dummy/empty one.